src/BlockActions.js

```diff
@@ -0,0 +1,181 @@
+    sortableBlock: PropTypes.bool,
+    canDuplicate: PropTypes.bool,
+    dragHandleRef: refType,
+  };
+
+  static defaultProps = {
+    sortableBlock: true,
+    canDuplicate: true,
+  };
+
+  get moveUpIcon() {
+    const {icons} = this.props;
+    if ((icons === undefined) || (icons.moveUp === undefined)) {
+      return <i className="fas fa-chevron-up" />;
+    }
+    return <span dangerouslySetInnerHTML={{__html: icons.moveUp}} />;
+  }
+
+  get moveDownIcon() {
+    const {icons} = this.props;
+    if ((icons === undefined) || (icons.moveDown === undefined)) {
+      return <i className="fas fa-chevron-down" />;
+    }
+    return <span dangerouslySetInnerHTML={{__html: icons.moveDown}} />;
+  }
+
+  get duplicateIcon() {
+    const {icons} = this.props;
+    if ((icons === undefined) || (icons.duplicate === undefined)) {
+      return <i className="fas fa-clone" />;
+    }
+    return <span dangerouslySetInnerHTML={{__html: icons.duplicate}} />;
+  }
+
+  get deleteIcon() {
+    const {icons} = this.props;
+    if ((icons === undefined) || (icons.delete === undefined)) {
+      return <i className="fas fa-trash" />;
+    }
+    return <span dangerouslySetInnerHTML={{__html: icons.delete}} />;
+  }
+
+  get isFirst() {
+    return this.props.index === 0;
+  }
+
+  get isLast() {
+    return this.props.index === (this.props.siblings.length - 1);
+  }
+
+  triggerCustomEvent(name, data=null) {
+    triggerCustomEvent(ReactDOM.findDOMNode(this), name, data);
+  }
+
+  sendKeyToDragHandle = key => {
+    const dragHandle = ReactDOM.findDOMNode(this.props.dragHandleRef.current);
+    triggerKeyboardEvent(dragHandle, 32);  // 32 for spacebar, to drag
+    return new Promise(resolve => {
+      setTimeout(() => {
+        triggerKeyboardEvent(dragHandle, key);
+        setTimeout(() => {
+          triggerKeyboardEvent(dragHandle, 32);  // Drop at the new position
+          resolve();
+        }, 100);  // 100 ms is the duration of a move in react-beautiful-dnd
+      }, 0);
+    });
+  };
+
+  moveUpHandler = event => {
+    event.preventDefault();
+    event.stopPropagation();
+    this.sendKeyToDragHandle(38)  // 38 for up arrow
+      .then(() => {
+        this.triggerCustomEvent('move', {index: this.props.index});
+      });
+  };
+
+  moveDownHandler = event => {
+    event.preventDefault();
+    event.stopPropagation();
+    this.sendKeyToDragHandle(40)  // 40 for down arrow
+      .then(() => {
+        this.triggerCustomEvent('move', {index: this.props.index});
+      });
+  };
+
+  duplicateHandler = event => {
+    event.preventDefault();
+    event.stopPropagation();
+    this.props.duplicateBlock();
+    this.triggerCustomEvent('duplicate');
+  };
+
+  deleteHandler = event => {
+    event.preventDefault();
+    event.stopPropagation();
+    this.props.hideBlock();
+  };
+
+  render() {
+    const {
+      blockDefinition, isSimpleLayout, sortableBlock, canDuplicate,
+    } = this.props;
+    return (
+      <aside>
```

Comment:
`aside` doesn't seem appropriate for UI elements like action buttons, and it should come with a class name anyway.